jetty-util/src/main/java/org/eclipse/jetty/util/BufferUtil.java

```diff
@@ -1096,15 +1094,15 @@ private static void appendDebugString(StringBuilder buf,ByteBuffer buffer)
     private static void appendContentChar(StringBuilder buf, byte b)
     {
         if (b == '\\')
-            buf.append("\\\\");   
-        else if (b >= ' ')
-            buf.append((char)b);
+            buf.append("\\\\");
         else if (b == '\r')
             buf.append("\\r");
         else if (b == '\n')
             buf.append("\\n");
         else if (b == '\t')
             buf.append("\\t");
+        else if ((b >= 0x20) && (b<=0x7E)) // limit to 7-bit printable US-ASCII character space
```

Comment:
This is probably entirely unnecessary, but I'd make this the first of the if-else sequence as it will be the most frequently executed so it will save nanoseconds :)

Reply:
wonder what the JIT will do to this ...

Reply:
it might be handled by the jit.  But then it might not and we might have to have the frequent path 4 branch predictions deep.    This is a debug method, so it's not really a problem but habitually I'd try to order such code to help JIT and/or branch prediction.

Reply:
I don't think so.   I think readable text OR hex is enough.  If anything I'd be inclined to remove the \n \r and just have them as hex.